fraud/testing.go

```diff
@@ -33,34 +35,33 @@ func (d *DummyService) Get(context.Context, ProofType) ([]Proof, error) {
 	return nil, nil
 }
 
-type mockStore struct {
-	headers    map[int64]*header.ExtendedHeader
-	headHeight int64
-}
+type H libheader.Header
+
+type mockStore mockstore.MockStore[H]
 
 // createStore creates a mock store and adds several random
 // headers.
 func createStore(t *testing.T, numHeaders int) *mockStore {
 	store := &mockStore{
-		headers:    make(map[int64]*header.ExtendedHeader),
-		headHeight: 0,
+		Headers:    make(map[int64]H),
+		HeadHeight: 0,
 	}
 
 	suite := headertest.NewTestSuite(t, numHeaders)
 
 	for i := 0; i < numHeaders; i++ {
 		header := suite.GenExtendedHeader()
-		store.headers[header.Height()] = header
+		store.Headers[header.Height()] = header
 
-		if header.Height() > store.headHeight {
-			store.headHeight = header.Height()
+		if header.Height() > store.HeadHeight {
+			store.HeadHeight = header.Height()
 		}
 	}
```

Comment:
We should use `mocks.NewStore` instead. It already implements header gen, when testsuite is passed to it as Generator.